### PR TITLE
use java8 language features like lambdas

### PR DESCRIPTION
**Pull request by spencergibb**
Compiles to java 7.

Animal Sniffer in [spring-cloud-build](https://github.com/spring-cloud/spring-cloud-build/blob/master/spring-cloud-build/pom.xml#L251-L271) keeps java 8 api's out of use.

What do you think @dsyer @mstine @joshlong @philwebb?

### CODE REVIEW

File: spring-cloud-consul-sample/src/main/java/org/springframework/cloud/consul/sample/SampleApplication.java [hunk 1]
```diff
@@ -82,7 +84,9 @@ public ServiceInstance me() {
 
 	@RequestMapping("/")
 	public ServiceInstance lb() {
-		return loadBalancer.choose(appName);
+		Function<String, ServiceInstance> f = loadBalancer::choose;
```
Comment (spencergibb) [inline spring-cloud-consul-sample/src/main/java/org/springframework/cloud/consul/sample/SampleApplication.java]: Using a method reference :-)

**Comment by spencergibb** [comment]
[Here](https://github.com/orfjackal/retrolambda#backported-language-features) are the supported language features.

**Comment by philwebb** [comment]
Looks interesting, but I'd approach with caution personally. I guess it depends how much benefit you get from the switch. I wonder if debugging is affected?

**Comment by dsyer** [comment]
Having gone through this with spring boot metrics, I would say it's worth the ugliness in our code for the guarantee that it definitely works for all our Java 7 users. We can add some Java 8 native samples - that might address the legitimate demands of the Java 8 constituency.

**Pull request closed without merging**